ext/dom/namednodemap.c

```diff
@@ -332,6 +336,27 @@ PHP_FUNCTION(dom_namednodemap_remove_named_item_ns)
 }
 /* }}} end dom_namednodemap_remove_named_item_ns */
 
+/* {{{ proto int|bool dom_namednodemap_count();
+*/
+PHP_FUNCTION(dom_namednodemap_count)
+{
+	zval *id;
+	dom_object *intern;
+	zval *count;
+
+	if (zend_parse_method_parameters(ZEND_NUM_ARGS(), getThis(), "O", &id, dom_namednodemap_class_entry) == FAILURE) {
+		return;
+	}
+
+	intern = Z_DOMOBJ_P(id);
+	if(dom_namednodemap_length_read(intern, &count)) {
+		RETURN_FALSE;
+	}
+
+	RETURN_LONG(count);
```

Comment:
What you probably want to do is just pass `return_value` as the second argument to `dom_namednodemap_length_read`.